stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -263,7 +263,7 @@ def _execute_search_pickings(self, domain, limit=None):
         )
 
     def _split_first_picking_for_limit(self, picking):
-        last_device = self.stock_device_type_ids[-1]
+        last_device = self.stock_device_type_ids.sorted()[-1]
```

Comment:
We could use a method to get the list of devices in order to centralize the required logic to ensure the right order etc...